src/main/java/dev/jbang/cli/Info.java

```diff
@@ -33,33 +61,77 @@
 
 		String mainClass;
 
+		List<String> dependencies;
+
+		List<Repo> repositories;
+
 		List<String> resolvedDependencies;
 
 		String javaVersion;
 
+		String requestedJavaVersion;
+
+		String availableJdkPath;
+
+		List<String> compileOptions;
+
 		List<String> runtimeOptions;
 
-		public ScriptInfo(Source src, RunContext ctx) {
-			String cp = ctx.resolveClassPath(src);
+		List<ResourceFile> files;
+
+		List<ScriptInfo> sources;
 
+		public ScriptInfo(Source src, RunContext ctx) {
 			originalResource = src.getResourceRef().getOriginalResource();
 			backingResource = src.getResourceRef().getFile().toString();
 
-			applicationJar = src.getJarFile().getAbsolutePath();
-			mainClass = ctx.getMainClassOr(src);
-
-			if (cp.isEmpty()) {
-				resolvedDependencies = Collections.emptyList();
-			} else {
-				resolvedDependencies = Arrays.asList(cp.split(CP_SEPARATOR));
+			ScriptSource ss = src.asScriptSource();
+			List<String> deps = ss.collectDependencies(System.getProperties());
+			if (!deps.isEmpty()) {
+				dependencies = deps;
 			}
-
-			if (ctx.getBuildJdk() > 0) {
-				javaVersion = Integer.toString(ctx.getBuildJdk());
+			if (!ss.collectRepositories().isEmpty()) {
+				repositories = ss	.collectRepositories()
+									.stream()
+									.map(repo -> new Repo(repo))
+									.collect(Collectors.toList());
+			}
+			List<RefTarget> refs = ss.collectFiles();
+			if (!refs.isEmpty()) {
+				files = refs.stream()
+							.map(ref -> new ResourceFile(ref))
+							.collect(Collectors.toList());
+			}
+			List<ScriptSource> srcs = ss.collectSources();
+			if (!srcs.isEmpty()) {
+				sources = srcs	.stream()
+								.map(s -> new ScriptInfo(s, null))
+								.collect(Collectors.toList());
+			}
```

Comment:
interesting indentation style ;-)

Reply:
That's @maxandersen enforcing his style with Spotless. I hate it myself but I squint my eyes when looking at the code and try not to see it :-D

Reply:
that explains it. Wikings are weird ;-)